compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -114,6 +114,16 @@ object RefChecks {
     }
   end checkSelfAgainstParents
 
+  /** warn if `@deprecatedInheritance` is present */
+  def warnDeprecatedInheritance(cls: ClassSymbol, parentTrees: List[Tree])(using Context): Unit =
+    val psyms = cls.parentSyms
+    for (psym, pos) <- psyms.zip(parentTrees.map(_.srcPos))
+        annot <- psym.getAnnotation(defn.DeprecatedInheritanceAnnot)
+    do
+      val msg = annot.argumentConstantString(0).getOrElse("")
+      val since = annot.argumentConstantString(1).getOrElse("")
+      report.deprecationWarning(em"inheritance from $psym is deprecated (since: $since): $msg", pos)
+
```

Comment:
No deprecation if an enclosing element (context owner) is deprecated; and the annot can have empty msg or since.
cf https://github.com/scala/scala/blob/2.13.x/src/compiler/scala/tools/nsc/typechecker/Typers.scala#L1859

Reply:
Noted. I actually moved the check from `RefChecks` to `CrossVersionChecks` too